internal/provider/resource_virtual_machine_group_test.go

```diff
@@ -0,0 +1,137 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/jimeh/undent"
+)
+
+func init() { //nolint:gochecknoinits
+	resource.AddTestSweepers("katapult_virtual_machine_group",
+		&resource.Sweeper{
+			Name: "katapult_virtual_machine_group",
+			F:    testSweepVMGroups,
+		})
+}
+
+func testSweepVMGroups(_ string) error {
+	m := sweepMeta()
+	ctx := context.TODO()
+
+	vmgs, _, err := m.Client.VirtualMachineGroups.List(
+		ctx, m.OrganizationRef(),
+	)
+	if err != nil {
+		return err
+	}
+
+	for _, vmg := range vmgs {
+		log.Printf(
+			"[DEBUG]  - Deleting Virtual Machine Group %s (%s)\n",
+			vmg.ID, vmg.Name,
+		)
+		_, err := m.Client.VirtualMachineGroups.Delete(ctx, vmg)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func TestAccKatapultVMGroup_basic(t *testing.T) {
+	tt := newTestTools(t)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: tt.ProviderFactories,
+		CheckDestroy:      testAccCheckKatapultVMGroupDestroy(tt),
+		Steps: []resource.TestStep{
+			{
+				Config: `resource "katapult_virtual_machine_group" "web" {
+					name = "test group"
+				}`,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(
```

Comment:
Should I also check the resource exists? Similar to this test: https://github.com/krystal/terraform-provider-katapult/blob/main/internal/provider/resource_ip_test.go#L104